dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifGoldenCopyIngestor.kt

```diff
@@ -94,6 +107,31 @@ class GleifGoldenCopyIngestor(
         logger.info("Finished processing of file $csvFile in ${getExecutionTime(start)}.")
     }
 
+    private fun prepareMappingFile() {
+        logger.info("Starting LEI-ISIN mapping update cycle for latest file.")
+        val tempFile = File.createTempFile("gleif_mapping_update", ".csv")
+        processMappingFile(tempFile, gleifApiAccessor::getIsinMappingFile)
+    }
+
+    @Synchronized
+    private fun processMappingFile(newMappingFile: File, downloadFile: (file: File) -> Unit) {
+        waitForBackend()
+        val start = System.nanoTime()
+
+        downloadFile(newMappingFile)
+        val newMappingFileCsv = gleifParser.getCsvFileFromZip(newMappingFile)
```

Comment:
Shouldn't this overwrite the savedMappingFile at some point? If that exists yet, where does that happen?

Reply:
replaceOldMappingFile in IsinDeltaBuilder.kt